### PR TITLE
Refactoring of ArtistsService

### PR DESCRIPTION
**Pull request by ReginaSchuster**
Renamed ArtistsService to ArtistService, factored out all search-related methods into new classes ArtistSearchService and ArtistSearchServiceImpl

### CODE REVIEW

File: webapp/src/test/java/rocks/metaldetector/web/controller/rest/ArtistsRestControllerTest.java [hunk 1]
```diff
@@ -73,7 +77,7 @@ void setUp() {
 
     @AfterEach
     void tearDown() {
-      reset(artistsService);
+      reset(artistSearchService);
```
Comment (DanielWagner87) [inline webapp/src/test/java/rocks/metaldetector/web/controller/rest/ArtistsRestControllerTest.java:80]: here we should reset all mocks of the test class.

File: webapp/src/test/java/rocks/metaldetector/service/artist/ArtistServiceImplTest.java [hunk 2]
```diff
@@ -45,7 +45,7 @@
 import static rocks.metaldetector.testutil.DtoFactory.SpotifyArtistSearchResultDtoFactory;
 
 @ExtendWith(MockitoExtension.class)
-class ArtistsServiceImplTest implements WithAssertions {
+class ArtistServiceImplTest implements WithAssertions {
```
Comment (DanielWagner87) [inline webapp/src/test/java/rocks/metaldetector/service/artist/ArtistServiceImplTest.java:48]: We should also split up the tests into `ArtistServiceImplTest` and `ArtistSearchServiceImplTest`. Convention is that we only test one class per test class and the instance of the class under test is called `underTest`

**Review by DanielWagner87 — COMMENTED** [review]
So far, so good. Two small things...

**Review by DanielW1987 — APPROVED** [review]
lgtm

**Pull request merged**